pkg/apis/ceph.rook.io/v1alpha1/types.go

```diff
@@ -170,13 +170,19 @@ type FilesystemList struct {
 // FilesystemSpec represents the spec of a file system
 type FilesystemSpec struct {
 	// The metadata pool settings
-	MetadataPool PoolSpec `json:"metadataPool"`
+	MetadataPool PoolSpec `json:"metadataPool,omitempty"`
 
 	// The data pool settings
-	DataPools []PoolSpec `json:"dataPools"`
+	DataPools []PoolSpec `json:"dataPools,omitempty"`
 
 	// The mds pod info
 	MetadataServer MetadataServerSpec `json:"metadataServer"`
+
+	// Do not create Ceph filesystem and pools.  Filesystem and Pools must have already been created in Ceph.
+	SkipPoolCreation bool `json:"skipPoolCreation"`
+
+	// Do not erase Ceph filesystem and pools when this filesystem is removed in Rook.
+	PreservePoolsOnRemove bool `json:"preservePoolsOnRemove"`
```

Comment:
At the time the pool management was implemented in Rook, there was some discussion about the lifetime of the pools and whether they should be deleted when the CRD was deleted. Without a confirmation flag such as `--yes-i-really-really-mean-it`, risk of deleting data unintentionally is higher. So even if rook creates the pools, it's good to have the separate flag that says to keep my pools.